src/Compilers/CSharp/Test/Emit2/Emit/RuntimeProbing/StackOverflowProbingTests.cs

```diff
@@ -0,0 +1,776 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests;
+
+public sealed class StackOverflowProbingTests : CSharpTestBase
+{
+#if NET
+    const string NativeIntDisplay = "nint";
+#else
+    const string NativeIntDisplay = "System.IntPtr";
+#endif
+
+    private static readonly EmitOptions s_emitOptions = EmitOptions.Default.WithInstrumentationKinds([InstrumentationKind.StackOverflowProbing]);
+
+    private CompilationVerifier CompileAndVerify(string source, string? expectedOutput = null, CSharpCompilationOptions? options = null, Verification? verification = null)
+        => CompileAndVerify(
+            source,
+            options: options ?? (expectedOutput != null ? TestOptions.UnsafeDebugExe : TestOptions.UnsafeDebugDll),
+            emitOptions: s_emitOptions,
+            verify: verification ?? Verification.Passes,
+            targetFramework: TargetFramework.NetLatest,
+            expectedOutput: expectedOutput);
+
+    private static void AssertNotInstrumented(CompilationVerifier verifier, string qualifiedMethodName)
+        => AssertInstrumented(verifier, qualifiedMethodName, expected: false);
+
+    private static void AssertInstrumented(CompilationVerifier verifier, string qualifiedMethodName, bool expected = true)
```

Comment:
>bool expected = true [](http://example.com/codeflow?start=101&length=20)

Is this parameter ever `true`?

Reply:
Oh, I thought it's in module cancellation... will fix